dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -166,10 +168,15 @@ protected void updateDBWorkflowWithSourceControlWorkflow(Workflow workflow, Work
             }
         }
 
+        // Remove versions that do not need to be updated
+        Set<WorkflowVersion> filteredVersions = newWorkflow.getWorkflowVersions().stream().filter(workflowVersion -> !Objects.equals(SKIP_COMMIT_ID, workflowVersion.getCommitID())).collect(
```

Comment:
I know I suggested the stream/filter, but by making it a Set there is the possibility that a version may get filtered out that wasn't being filtered before. Probably never happens with our data, but still... I would just get rid of the set and the intermediate variable, and use forEach directly on the stream:

```
newWorkflow.getWorkflowVersions().stream()
    .filter(workflowVersion -> !Objects.equals(SKIP_COMMIT_ID, workflowVersion.getCommitID()))
    .forEach(version -> {
        WorkflowVersion workflowVersionFromDB = existingVersionMap.get(version.getName());
        ...
});